src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx

```diff
@@ -44,16 +46,27 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
 }) => {
   const { router } = useRouter()
   const { sendToast } = useToasts()
-  const initialValues = getMyCollectionArtworkFormInitialValues(artwork)
-  const [shouldShowModal, setShouldShowModal] = useState<boolean>(false)
+  const { createOrUpdateArtwork } = useCreateOrUpdateArtwork()
+  const { submitMutation: deleteArtworkImage } = useDeleteArtworkImage()
 
+  const initialValues = getMyCollectionArtworkFormInitialValues(artwork)
   const initialErrors = validateArtwork(
     initialValues,
     MyCollectionArtworkDetailsValidationSchema
   )
-  const { createOrUpdateArtwork } = useCreateOrUpdateArtwork()
+
+  const [shouldShowModal, setShouldShowModal] = useState<boolean>(false)
 
   const handleSubmit = async (values: ArtworkModel) => {
+    // Set edition values for attribution class
+
+    if (values.attributionClass !== "LIMITED_EDITION") {
+      values.editionNumber = ""
+      values.editionSize = ""
+    }
```

Comment:
Do you mean something like `adjustValues`? I'm not sure about it because the values are getting adjusted further down when updating the artwork as well.

Reply:
ah oki, i didn't pay attention to that. But `adjustValues` is exactly what I meant

Reply:
What do you think? Shall we move all the logic into an extra function or just leave it here in `handleSubmit`?